cloudconfig/instancecfg/instancecfg.go

```diff
@@ -438,6 +439,19 @@ func (cfg *InstanceConfig) APIHostAddrs() []string {
 	return hosts
 }
 
+func (cfg *InstanceConfig) APIHosts() []string {
+	var hosts []string
+	if cfg.Bootstrap != nil {
+		hosts = append(hosts, "localhost")
+	}
+	if cfg.APIInfo != nil {
+		for _, addr := range cfg.APIInfo.Addrs {
+			hosts = append(hosts, strings.Split(addr, ":")[0])
```

Comment:
strings.Split isn't sufficient for getting the host part of an address when hosts can be IPv6.
I think you probably want net.SplitHostPort instead.